app/src/main/java/com/quarkworks/apartmentgroceries/service/SyncGroceryItem.java

```diff
@@ -38,13 +48,30 @@ public void done(@Nullable JSONObject jsonObject) {
                 try {
 
                     JSONArray groceryJsonArray = jsonObject.optJSONArray(JsonKeys.RESULTS);
-
-                    for (int i = 0; i < groceryJsonArray.length(); i++) {
-                        RGroceryItem groceryItem = realm.createObject(RGroceryItem.class);
-                        groceryItem.setName(groceryJsonArray.getJSONObject(i).optString(JsonKeys.NAME));
+                    if (groceryJsonArray != null) {
```

Comment:
I didn't explain this very well the first time, but generally we want to use JSONObject.opt() methods when we don't know if something will be there (like for the purchasedBy column) and JSONObject.get() methods when we expect something will be there.

In this case, we expect that there will be an array there, so we should you JSONObject.getJSONArray(). If that isn't there, we've done something very wrong and we need to log an error.